### ISSUE TITLE
Problems with test results for original and updated code

### ISSUE THREAD

**Issue by tjnkyqcy** [Ipsedo/MARLClassification#11]
Thank you for the "MARLClassification-original-paper", luckily the code is working without much modification. But I still have a question, I compared the code of "MARLClassification-original-paper" with the code of "MARLClassification (code with criterion)". " codes with the training parameters shown below.
-a 3 --step 5 --cuda --run-id train_mnist train --action [[1,0],[-1,0],[0,1],[0,-1]] --img-size 28 --nb-class 10 -d 2 --f 6 --ft-extr mnist --nb 64 --na 64 --nm 16 --nd 8 --nlb 96 --nla 96 --batch-size 32 --lr 1e-3 --nb-epoch 40 --eps 1. --eps-dec 0.99995 -o ./out/mnist

I compared the train_prec and eval_prec parameters and didn't find that using action-critic gives better results as you said, here are the comparison curves for the two methods.
![eval_prec](https://github.com/user-attachments/assets/5ead55c5-c027-4e4e-acf1-4d6613ae67f4)
![train_prec](https://github.com/user-attachments/assets/ddb25922-5190-4df1-9dbf-83eab6b1d9b4)

I'm wondering if there's a problem with my settings somewhere? Thanks.

**Comment by tjnkyqcy** [comment]
The test data used is the valid data in mnist. original-paper and update test set results are shown below
![test(original-paper)](https://github.com/user-attachments/assets/f34eeb5c-c46d-4269-b72a-6b8d3df1eb08)
![test(update)](https://github.com/user-attachments/assets/6160634b-1e6d-4542-a1b6-5594a7d9fa87)
Using the actor-critic method in the test set didn't work as well as before either.

**Comment by Ipsedo** [comment]
I no longer benchmark on this dataset as it is "too easy" to train and not relevant in my opinion.
Maybe you can try other datasets like Resisc-45 or AID. 

Also movements of the agents were more natural with actor critic. If I remember well, with original version, some combinations of hyper-parameters were resulting in a same action for each agent at every steps.

**Comment by Ipsedo** [comment]
Any update from your side? If not, can I close this issue?
